sbin/root-mount

```diff
@@ -55,15 +57,14 @@ LVM_RESERVED_KB=$(expr $LVM_RESERVED_MB \* 1024)
 # This function should not fail even if the operations cannot be performed
 check_firstboot_resize()
 {
-	FREE_EXTENTS=$(lvm vgdisplay sailfish -c | cut -d ":" -f 16)
-	EXTENT_SIZE=$(lvm vgdisplay sailfish -c | cut -d ":" -f 13)
-	FREE_KB=$(expr $FREE_EXTENTS \* $EXTENT_SIZE)
-
-	if test "$FREE_KB" -le "$LVM_RESERVED_KB"; then
+	if test "$IS_FS_RESIZED" -eq 1; then
 		# The space is allocated, nothing to do.
 		return 0
 	fi
 
+	FREE_EXTENTS=$(lvm vgdisplay sailfish -c | cut -d ":" -f 16)
+	EXTENT_SIZE=$(lvm vgdisplay sailfish -c | cut -d ":" -f 13)
+
 	# Increase root size
 	if lvm lvextend --size "$LVM_ROOT_SIZE"M "$ROOTDEV"; then
 		e2fsck -f -y "$ROOTDEV" > /dev/kmsg
```

Comment:
If device reboots here between e2fsck and resize2fs, then lvm is extended and filesystem is not. So does the code come here on next loop?